### DISCUSSION TITLE
Multiple problems with [@types/mongoose-delete]

### DISCUSSION THREAD

**Discussion by Paso** [DefinitelyTyped/DefinitelyTyped#68603] [Issues with a @types package]
The @types/mongoose-delete package has a few problems.

1) The `count`-method has been removed from mongoose 8.x
2) The `SoftDeleteModel` generic is missing arguments from the underlying `mongoose.Model`
3) The `plugin` interface is broken, see issue [mongoose-delete#151](https://github.com/dsanel/mongoose-delete/issues/151)

I have been unable to understand 3) but can work around it by doing 
```
(schema as unknown as mongoose.Schema).plugin(...)
```

[a comment by an automated account is not shown]

**Comment by alexbrazier** [comment]
This PR should fix the first two issues https://github.com/DefinitelyTyped/DefinitelyTyped/pull/68633, not sure about the 3rd.

**Reply by Paso** [reply]
The PR fixes 1. but not 2.

Is there perhaps a different PR for that issue? As far as I can tell the fix is just adding the generic arguments and passing them through.

**Reply by alexbrazier** [reply]
You're right, I think I misunderstood, but I assume this is what you mean? Not sure if there is an easier way of doing it without having to type everything out and update as mongoose changes? 

```ts
interface SoftDeleteModel<TRawDocType,
    TQueryHelpers = {},
    TInstanceMethods = {},
    TVirtuals = {},
    THydratedDocumentType = mongoose.HydratedDocument<TRawDocType, TVirtuals & TInstanceMethods, TQueryHelpers>,
    TSchema = any>
        extends mongoose.Model<TRawDocType,
        TQueryHelpers,
        TInstanceMethods,
        TVirtuals,
        THydratedDocumentType,
        TSchema>
    {
```

**Reply by Paso** [reply]
Yes, that is what I had in mind, but I wouldn't be surprised if there is som way of reusing the types from mongoose that only the greyest of typescript wizards know of.

**Reply by Paso** [reply]
@alexbrazier Is there something blocking this change? Anything I can help with?

**Reply by alexbrazier** [reply]
@Paso I just submitted the last PR as I needed the update, but I didn't create or maintain the original package. Best way to get the changes would be to submit a new PR to [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped) with the changes you need. I'd be happy to review it.

**Reply by facularrauri** [reply]
@Paso @alexbrazier I just added this PR to update SoftDeleteModel [#69633]
